src/notation/internal/masternotation.cpp

```diff
@@ -356,6 +356,9 @@ void MasterNotation::applyOptions(mu::engraving::MasterScore* score, const Score
                 score->style().set(st, DefaultStyle::defaultStyle().value(st));
             }
         }
+
+        // the new score should get the staffDefaultMergeMatchingRests from the default settings rather than the template
+        score->style().set(Sid::staffDefaultMergeMatchingRests, DefaultStyle::defaultStyle().value(Sid::staffDefaultMergeMatchingRests));
```

Comment:
If the templates are ever saved in MuseScore 4.5 (with this PR included) then they will have the style setting and override the default. (I tested this, which is why I added this code.) To me, this style setting should override the templates.

Reply:
It will also override the setting in a user-created template. I guess it's a philosophy of what elements of the user's default style sheet should override what's in the template. I don't have a clear answer in my head, and if you say, "None", then I'll take it out. It isn't as big of a deal now that we are using `AutoOnOff`.

Reply:
I'm still not very confident about this, but I think we should take it out, because:
- if the user explicitly sets this to true in a template, and then reuses that template, it's reasonable to expect that the value from the template will be taken
- currently, we're not applying this treatment to any other style setting, except to page settings which are of course a bit special. To keep behaviour consistent, I wouldn't be in favour of adding dedicated logic for this specific setting.

Reply:
If this becomes an issue, I could see a "use template styles" vs "use default styles" switch somewhere as a better, more complete solution.

Reply:
Removed it.